Remotion/Mixins/UnitTests/Core/CodeGeneration/AttributeBasedMetadataImporterTest.cs

```diff
@@ -368,5 +370,27 @@ private Type CreateTypeWithFakeWrappers ()
 
       return wrapperClassBuilder.CreateType ();
     }
+
+    private class ClassWithNoAttributes
+    {
+    }
+
+    [ConcreteMixedType (new object[] { typeof (object), new object[0], new Type[0] }, new Type[0])]
+    private class ClassWithConcreteMixedTypeAttribute
+    {
+    }
+
+    [ConcreteMixinType (new object[] { typeof (object), new object[0], new object[0] })]
+    private class ClassWithConcreteMixinTypeAttribute
+    {
+    }
+
+    private class DerivedClassWithConcreteMixedTypeAttribute : ClassWithConcreteMixedTypeAttribute
```

Comment:
```suggestion
    private class DerivedClassWithConcreteMixedTypeAttributeOnBaseClass : ClassWithConcreteMixedTypeAttribute
```

Naming: the original name was ambigous, we don't know from the class name that the attribute would be on the base.
We could use `DerivedClassForBaseClassWith...Attribute` or the suggestion from the inline edit. I think the second option from the explanation might be more correct but probably less readable? Dealer's choice. The second class also needs a rename then.